From 64a64f5ef90dfb4b3b862d0943437f11e35db938 Mon Sep 17 00:00:00 2001
From: Maarten Coene <maartenc@apache.org>
Date: Thu, 31 Jan 2013 22:38:20 +0000
Subject: [PATCH] FIX: NullPointerException when using httpclient and server
 doesn't return content-type header (IVY-1400) (thanks to Frederic Riviere)

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@1441235 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES.txt                                             | 2 ++
 src/java/org/apache/ivy/util/url/HttpClientHandler.java | 5 ++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/CHANGES.txt b/CHANGES.txt
index 46a476640..30a405a4f 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -105,6 +105,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 	Geoff Reedy
 	Torkild U. Resheim
 	Christian Riege
+	Frederic Riviere
 	Jens Rohloff
 	Andreas Sahlbach
 	Brian Sanders
@@ -142,6 +143,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 - IMPROVEMENT: add support for source bundles from p2 repositories
 - IMPROVEMENT: add support for source URI from OBR repositories
 
+- FIX: NullPointerException when using httpclient and server doesn't return content-type header (IVY-1400) (thanks to Frederic Riviere)
 - FIX: Properly handle evicted nodes in ResolveReport
 
    2.3.0
diff --git a/src/java/org/apache/ivy/util/url/HttpClientHandler.java b/src/java/org/apache/ivy/util/url/HttpClientHandler.java
index 00bbc6a58..64d225665 100644
--- a/src/java/org/apache/ivy/util/url/HttpClientHandler.java
+++ b/src/java/org/apache/ivy/util/url/HttpClientHandler.java
@@ -153,9 +153,8 @@ public URLInfo getURLInfo(URL url, int timeout) {
                 method = doGet(url, timeout);
             }
             if (checkStatusCode(url, method)) {
-                String contentType = method.getResponseHeader("content-type").getValue();
-                String bodyCharset = BasicURLHandler.getCharSetFromContentType(contentType);
-                return new URLInfo(true, getResponseContentLength(method), getLastModified(method), bodyCharset);
+                return new URLInfo(true, getResponseContentLength(method), 
+                        getLastModified(method), method.getRequestCharSet());
             }
         } catch (HttpException e) {
             Message.error("HttpClientHandler: " + e.getMessage() + ":" + e.getReasonCode() + "="
